src/model/channel/message.rs

```diff
@@ -151,6 +151,41 @@ pub struct Message {
     pub poll: Option<Box<Poll>>,
 }
 
+// Custom deserialize function to deserialize components safely without knocking the whole message
+// out when new components are found but not supported.
+fn deserialize_components<'de, D>(deserializer: D) -> Result<Vec<ActionRow>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct ComponentsVisitor;
+
+    impl<'de> Visitor<'de> for ComponentsVisitor {
+        type Value = Vec<ActionRow>;
+
+        fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            formatter.write_str("a sequence of ActionRow elements")
+        }
+
+        fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+        where
+            A: serde::de::SeqAccess<'de>,
+        {
+            let mut components = Vec::new();
+            while let Some(action_row) = seq.next_element::<serde_json::Value>()? {
+                match serde_json::from_value::<ActionRow>(action_row) {
```

Comment:
Using serde_json::Value here seems like a bad idea, is there a way to use `flatten` and `RawValue` to only deserialize the type field and whitelist type values?

Reply:
Should be better now?